### PR TITLE
Geth test integration

### LINKED ISSUE #1089: Run geth with tests (`mix test --geth`)

### PR DESCRIPTION
**Pull request by mederic-p**
Issue/Task Number: 1089
Closes #1089 

# Overview

This PR adds a few integration tests using `geth`.
Geth must be installed on the machine to successfully run these tests.
Note that I haven't added/changed anything to circle ci pipeline so it won't run these integration tests.

# Changes

- Add a few integration tests.

# Implementation Details

A geth node will be started for each integration test using the `--dev` option.
Each test takes about 7 sec to run.

# Usage

The existing tests can be ran as usual with `mix test` and the new integration tests can be ran using `mix test --only integration`

### CODE REVIEW

File: apps/eth_blockchain/test/support/wait_for.ex [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+# limitations under the License.
+
+defmodule EthBlockchain.WaitFor do
+  @moduledoc """
+  Generic wait_for_* utils, styled after web3 counterparts
+  """
+
+  alias Ethereumex.HttpClient
+
+  def eth_rpc do
```
Comment (unnawut) [inline apps/eth_blockchain/test/support/wait_for.ex]:
```suggestion
  def eth_rpc(timeout \\ 10_000) do
```

For consistency with `eth_receipt/2`

File: apps/eth_blockchain/test/eth_blockchain/integration/contract_test.exs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          name: "OMGToken",
+          symbol: "OMG",
+          decimals: 18,
+          initial_amount: 100_000
+        })
+
+      assert res == :ok
+      assert tx_hash != nil
+      assert contract_address != nil
+      assert contract_uuid == "3681491a-e8d0-4219-a40a-53d9a47fe64a"
```
Comment (unnawut) [inline apps/eth_blockchain/test/eth_blockchain/integration/contract_test.exs]: Can this be refactored into a shared function somewhere called `locked_comtract_uuid()` or similar?

File: apps/eth_blockchain/test/eth_blockchain/integration/contract_test.exs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          name: "OMGToken",
+          symbol: "OMG",
+          decimals: 18,
+          initial_amount: 100_000
+        })
+
+      assert res == :ok
+      assert tx_hash != nil
+      assert contract_address != nil
+      assert contract_uuid == "9e0340c0-9aa4-4a01-b280-d400bc2dca73"
```
Comment (unnawut) [inline apps/eth_blockchain/test/eth_blockchain/integration/contract_test.exs]: Can this be refactored into a shared function somewhere called `unlocked_comtract_uuid()` or similar?

File: apps/eth_blockchain/test/support/dumb_subscriber.ex [hunk 4]
```diff
@@ -35,6 +35,7 @@ defmodule EthBlockchain.DumbSubscriber do
       state
       |> Map.put(:receipt, receipt)
       |> Map.put(:confirmations_count, confirmations_count)
+      |> Map.put(:error, nil)
```
[the remark this answers is not in the document]
Comment (mederic-p) [reply apps/eth_blockchain/test/support/dumb_subscriber.ex]: I'll refactor this file a bit to make it more understandable

File: apps/eth_blockchain/test/eth_blockchain/integration/contract_test.exs (hunk 2 above)
Comment (mederic-p) [reply apps/eth_blockchain/test/eth_blockchain/integration/contract_test.exs]: sure

**Comment by sirn** [comment]
For the CI pipeline, I think the easiest way to do it is to attach Geth as a sidecar in `executors.builder_pg` (probably rename it to `builder_intg` or something), e.g. something among the line of:

```yaml
- image: ethereum/client-go:release-1.9
  entrypoint:
    - geth
    - --dev
```

Default Geth Dockerfile should expose port 8545, 8546 and 30303 by default, so you can access it via localhost:8545, etc. in the `test` step. E2E on the other hand rely on `docker-compose` which you can just add geth as another container in `docker-compose`.

**Pull request merged**